### PR TITLE
embed class

### PR DESCRIPTION
**Pull request by Killermaschine88**

### CODE REVIEW

**Review by twisttaan — APPROVED** [review]
looks good to me
cc @Nciklol

**Comment by thehairy** [comment]
How about finishing it properly and then making a PR? XD

**Comment by twisttaan** [comment]
> How about finishing it properly and then making a PR? XD

I mean at least it's a start, though that is a good point. I was just being open-minded because of how new of a project this is.

**Comment by thehairy** [comment]
why do you close it, he can make it a draft :D

File: src/structures/Embed.ts [hunk 1]
```diff
@@ -1,25 +1,126 @@
...
+        this.timestamp = null
+        this.color = null
+        this.footer = null
+        this.image = null
+        this.thumbnail = null
+        this.video = null
+        this.provider = null
+        this.author = null
+        this.fields = []
 	}
```
Comment (megatank58) [inline src/structures/Embed.ts:29]: constructor should support passing EmbedOptions and setting it directly

**Comment by twisttaan** [comment]
oh right, new to this 😅

File: src/structures/Embed.ts [hunk 2]
```diff
@@ -1,25 +1,126 @@
...
+        return this.description = description
+    }
+
+    public setURL(description: String | null) {
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:53]: URL should be setted here, not description

File: src/structures/Embed.ts [hunk 3]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setTimestamp(timestamp: Boolean | null) {
+        if(typeof timestamp === null) {
+            return this.description = null
+        }
```
Comment (megatank58) [inline src/structures/Embed.ts:58]: timestamp, not description

File: src/structures/Embed.ts [hunk 4]
```diff
@@ -1,25 +1,126 @@
...
+        return this.color = color //unfinished needs a way to set the correct format
+    }
+
+    public setFooter(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:77]: footer, not description

File: src/structures/Embed.ts [hunk 5]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setImage(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:85]: image, not description

File: src/structures/Embed.ts [hunk 6]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setThumbnail(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:93]: thumbnail, not description

File: src/structures/Embed.ts [hunk 7]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setVideo(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:101]: video, not description

File: src/structures/Embed.ts [hunk 8]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setProvider(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:109]: provider, not description

File: src/structures/Embed.ts [hunk 9]
```diff
@@ -1,25 +1,126 @@
...
+
+    public setAuthor(description: String | null) { //unfinished
+        if(typeof description === null) {
+            return this.description = null
+        }
+        if(typeof description !== "string") throw new Error(`Expected type string but got ${typeof description} for embed description`)
+        return this.description = description
+    }
```
Comment (megatank58) [inline src/structures/Embed.ts:117]: author not description

File: src/structures/Embed.ts [hunk 10]
```diff
@@ -1,25 +1,126 @@
...
+
+    public addField(description: String | null) { //unfinished
```
Comment (megatank58) [inline src/structures/Embed.ts:119]: might wanna make this addFields and have multiple fields, also not description

**Review by megatank58 — CHANGES_REQUESTED** [review]

**Comment by Killermaschine88** [comment]
as i said i havent finished it lol

File: src/structures/Embed.ts (hunk 10 above)
Comment (Nciklol) [reply src/structures/Embed.ts:119]: > might wanna make this addFields and have multiple fields

Why?

**Comment by Nciklol** [comment]
This PR should be marked as a draft.

**Comment by Killermaschine88** [comment]
> This PR should be marked as a draft.

how

**Pull request closed without merging**